### PR TITLE
[CCI] Create advanced index actions guide

### LINKED ISSUE #140: [Guide] Advanced Index Actions
Create `guides/advanced_index_actions.md` similar to the [Ruby guide](https://github.com/opensearch-project/opensearch-ruby/blob/main/guides/advanced_index_actions.md)

You must assure that: 
- All code-blocks, when run from start to finish in order they appear in the guide, do not throw any errors. If you want to demonstrate that certain statements will throw errors, they must be wrapped in a `try/catch` and print out the expected error.
- All code-blocks are properly marked with the language it's written in (i.e.  ` ```ruby `)
- The instructions, though mostly identical to the Ruby guide, are suitable for this repo.

### PR DESCRIPTION
**Pull request by Nicksqain**
### Description

- [x] Create advanced index actions guide

Using the doc: [Opensearch rust](https://docs.rs/opensearch/latest/opensearch/)

### Issues Resolved
- closes #140 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Xtansia — CHANGES_REQUESTED** [review]
Your guide is still referencing and using Ruby, it has not been updated to Rust

**Comment by dblock** [comment]
- Let's link this new guide from USER_GUIDES?
- Link checker is complaining.

I think we already are discussing how to run these samples in CI, separate issue but thought I'd mention it.

**Comment by Nicksqain** [comment]
@dblock 
Check if this is how you want to add the link?
The way I see it is that there will be links to all the guides in the file. And in the new guide I can add a link to .rs, where you can see how to import and use the code.

File: USER_GUIDE.md [hunk 1]
```diff
@@ -11,6 +12,12 @@
 
 # User Guide
 
+## All guides
+
+We advise you to read all the manuals that relate to one part or another of your work with the client:
```
Comment (dblock) [inline USER_GUIDE.md:17]: I would remove this, feels like we're saying something obvious ;)

File: USER_GUIDE.md [hunk 2]
```diff
@@ -11,6 +12,12 @@
...
+
+- [Advanced index actions](guides/advanced_index_actions.md)
```
Comment (dblock) [inline USER_GUIDE.md:19]: Advanced Index Actions

File: guides/advanced_index_actions.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
+# Advanced Index Actions
+
+In this guide, we will look at some advanced index actions that are not covered in the [Index Lifecycle](index_lifecycle.md) guide.
```
Comment (dblock) [inline guides/advanced_index_actions.md:3]: That file doesn't exist yet, so the link checker breaks. You can combine all the guides in 1 PR or make them in order they link to each-other.

**Review by dblock — COMMENTED** [review]
I like it. Link checker needs to pass.

Consider linking to the documentation concepts like "force merge" when they are mentioned.

File: guides/advanced_index_actions.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+Let's create a client instance, and an index named `movies`:
+
+```rust
+// Create a client to make API calls to OpenSearch running on http://localhost:9200.
+let client = OpenSearch::default();
+// Alternatively, you can create a client to make API calls against OpenSearch running on a specific url::Url.
+let url = Url::parse("https://example.com")?;
+let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url)).cert_validation(CertificateValidation::None).build()?;
+let client = OpenSearch::new(transport);
```
Comment (Xtansia) [inline guides/advanced_index_actions.md:15]: Let's make this consistent with the other guides:
```suggestion
// Create a client to make API calls to OpenSearch running on https://localhost:9200.
let url = Url::parse("https://localhost:9200")?;
let credentials = Credentials::Basic("admin".into(), "admin".into());
let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url))
      .cert_validation(CertificateValidation::None)
      .auth(credentials)
      .build()?;
let client = OpenSearch::new(transport);
```

File: guides/advanced_index_actions.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+```rust
+client.indices().clear_cache(IndicesClearCacheParts::Index(&["movies"])).send().await?;
+```
+
+By default, the `indices.clear_cache` API action clears all types of cache. To clear specific types of cache pass the the `query`, `fielddata`, or `request` parameter to the API action:
+
+```rust
+client.indices().clear_cache(IndicesClearCacheParts::Index(&["movies"])).query(true).send().await?;
+    client.indices().clear_cache(IndicesClearCacheParts::Index(&["movies"])).fielddata(true).request(true).send().await?;
```
Comment (Xtansia) [inline guides/advanced_index_actions.md:33]: The indentation is a little weird here, let's fix it like so:
```suggestion
client.indices().clear_cache(IndicesClearCacheParts::Index(&["movies"])).query(true).send().await?;
client.indices().clear_cache(IndicesClearCacheParts::Index(&["movies"])).fielddata(true).request(true).send().await?;
```

File: guides/advanced_index_actions.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+```rust
+client.indices().add_block(IndicesAddBlockParts::IndexBlock(&["movies"],"write")).send().await?;
+client.indices().clone(IndicesCloneParts::IndexTarget("movies","movies_clone")).send().await?;
+client.indices().put_settings(IndicesPutSettingsParts::Index(&["movies"]))
+    .body(json!({
+    "index": {
+        "blocks": {
+            "write": false
+        }
+    }
```
Comment (Xtansia) [inline guides/advanced_index_actions.md:82]:
```suggestion
        "index": {
            "blocks": {
                "write": false
            }
        }
```

File: guides/advanced_index_actions.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    })).send().await?;
+```
+
+### Split index
+
+You can split an index into another index with more primary shards. The source index must be in read-only state for splitting. The following example create the read-only `books` index with 30 routing shards and 5 shards (which is divisible by 30), splits index into `bigger_books` with 10 shards (which is also divisible by 30), then re-enables write:
```
Comment (Xtansia) [inline guides/advanced_index_actions.md:88]:
```suggestion
You can split an index into another index with more primary shards. The source index must be in read-only state for splitting. The following example creates the read-only `books` index with 30 routing shards and 5 shards (which 30 is divisible by), splits the index into `bigger_books` with 10 shards (which 30 is also divisible by), then re-enables writes:
```

File: guides/advanced_index_actions.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+    .body(json!({
+    "settings": {
+        "index": {
+            "number_of_shards": 5,
+            "number_of_routing_shards": 30,
+            "blocks": {
+                "write": true
+            }
+        }
+    }
```
Comment (Xtansia) [inline guides/advanced_index_actions.md:101]:
```suggestion
        "settings": {
            "index": {
                "number_of_shards": 5,
                "number_of_routing_shards": 30,
                "blocks": {
                    "write": true
                }
            }
        }
```

File: opensearch/examples/advanced_index_actions.rs [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Let's create a client instance, and an index named `movies`:
+
+    // Create a client to make API calls to OpenSearch running on http://localhost:9200.
+    let _client = OpenSearch::default();
+    // Alternatively, you can create a client to make API calls against OpenSearch running on a specific url::Url.
+    let url = Url::parse("https://example.com")?;
+    let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url)).cert_validation(CertificateValidation::None).build()?;
+    let client = OpenSearch::new(transport);
```
Comment (Xtansia) [inline opensearch/examples/advanced_index_actions.rs:15]: Let's make this consistent with the other guides:
```suggestion
    // Create a client to make API calls to OpenSearch running on https://localhost:9200.
    let url = Url::parse("https://localhost:9200")?;
    let credentials = Credentials::Basic("admin".into(), "admin".into());
    let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url))
          .cert_validation(CertificateValidation::None)
          .auth(credentials)
          .build()?;
    let client = OpenSearch::new(transport);
```

File: opensearch/examples/advanced_index_actions.rs [hunk 10]
```diff
@@ -0,0 +1,60 @@
+use opensearch::{ http::transport::TransportBuilder, OpenSearch, http::transport::SingleNodeConnectionPool, cert::CertificateValidation};
```
Comment (Xtansia) [inline opensearch/examples/advanced_index_actions.rs:1]: Please run `cargo fmt`

File: opensearch/examples/advanced_index_actions.rs [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    client.indices().clone(IndicesCloneParts::IndexTarget("movies","movies_clone")).send().await?;
+    client.indices().put_settings(IndicesPutSettingsParts::Index(&["movies"]))
+      .body(json!({
+        "index": {
+            "blocks": {
+                "write": false
+            }
+        }
+      })).send().await?;
+    // You can split an index into another index with more primary shards. The source index must be in read-only state for splitting. The following example create the read-only `books` index with 30 routing shards and 5 shards (which is divisible by 30), splits index into `bigger_books` with 10 shards (which is also divisible by 30), then re-enables write:
```
Comment (Xtansia) [inline opensearch/examples/advanced_index_actions.rs:42]:
```suggestion
    // You can split an index into another index with more primary shards. The source index must be in read-only state for splitting. The following example creates the read-only `books` index with 30 routing shards and 5 shards (which 30 is divisible by), splits the index into `bigger_books` with 10 shards (which 30 is also divisible by), then re-enables writes:
```

File: guides/advanced_index_actions.md (hunk 3 above)
Comment (Nicksqain) [reply guides/advanced_index_actions.md:3]: @dblock  Can we wait for the index lifecycle to merge first and then re-run the link checker?

**Review by Xtansia — APPROVED** [review]

**Pull request merged**